### PR TITLE
Slow rollouts

### PR DESCRIPTION
**Pull request by mrrrgn**
This is just to test the waters. The changes here only impact spot requests. Moving the changes to on-demand instances will require a good deal more refactoring.

### CODE REVIEW

File: cloudtools/scripts/aws_watch_pending.py [hunk 1]
```diff
@@ -38,6 +38,68 @@
...
+    """
+    Uses the current ratio of previous/latest ami instaces, a target
+    "latest" percentage, and a number of new instances to be launched
+    to calculate a distribution of new amis which would move
+    the overall ratio closer to the target.
+
+    e.g. if we have 60 instances running our old ami and 40 instances
+    running latest, and we wish to launch another 10 instances with the
+    same 60/40 split, the function would return (4, 6) to indicate that
+    4 previous and 6 latest amis should be launched.
```
Comment (djmitche) [inline cloudtools/scripts/aws_watch_pending.py]: Same algorithm as dollar cost averaging

**Comment by djmitche** [comment]
So this determines the "previous" AMI based on what's already running?

**Comment by mrrrgn** [comment]
It pulls down images which aren't running too. The boto function at the base of everything is get_all_images

File: cloudtools/scripts/aws_watch_pending.py [hunk 2]
```diff
@@ -38,6 +38,68 @@
...
+    (0, 40)
+    >>> find_prev_latest_amis_needed(0, 10, 10, 40)
+    (40, 0)
+    >>> find_prev_latest_amis_needed(30, 0, 7, 3)
+    (0, 3)
+    >>> find_prev_latest_amis_needed(30, 7, 0, 3)
+    (3, 0)
+    """
+    total_amis = prev_ami_count + latest_ami_count
+    if latest_ami_percentage < 100 and latest_ami_percentage >= 0:
```
Comment (rail) [inline cloudtools/scripts/aws_watch_pending.py]: In Python you can 
```python
if 0 <= latest_ami_percentage < 100:
    ...
```
:p

**Comment by rail** [comment]
LGTM

**Pull request merged**